tests/lib/rules/no-unnecessary-act.test.ts

```diff
@@ -0,0 +1,782 @@
+import { createRuleTester } from '../test-utils';
+import rule, { RULE_NAME } from '../../../lib/rules/no-unnecessary-act';
+
+const ruleTester = createRuleTester();
+
+/**
+ * - AGR stands for Aggressive Reporting
+ * - RTL stands for React Testing Library (@testing-library/react)
+ * - RTU stands for React Test Utils (react-dom/test-utils)
+ */
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    {
+      code: `// case: RTL act wrapping non-RTL calls
```

Comment:
seeing this code makes me wonder if it is worth (not for this PR tho) adding a property to the test case that contains a "description" or something (that will be added as the test description when running)

Reply:
I've tried to do that in the past but miserably failed. Definitely interested in giving this a try again, but I think ESLint itself is working into adding this feature.

Reply:
I think this should work https://eslint.org/docs/developer-guide/nodejs-api#customizing-ruletester

what do you think? I can create a ticket and work on it at some point in the next week (after all I don't think this is high priority so it matches my current free time haha)

Reply:
I'm assuming the method parameter will have the info I need 😅

Reply:
That's the customization I tried to do in the past but didn't work. It was back when still using ESLint v6 so maybe it's possible now (or I just didn't find how to do it properly). Sounds like a plan! Create the issue and give it a try 👌

Reply:
yeah I went to their docs and there's no useful information to retrieve it https://github.com/eslint/eslint/blob/c981fb1994cd04914042ced1980aa86b68ba7be9/lib/rule-tester/rule-tester.js#L881 here is what it is being called.

I think overriding `describe` in [here](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/tests/lib/test-utils.ts#L31) just before calling `super.run` would allow me to iterate the rules and call the tests ( I have the reference from the `run` method. I would essentially re-implementing those ~10 lines from eslint test runner but it should work